Shared/Samples/Edit with branch versioning/EditWithBranchVersioningView.Views.swift

```diff
@@ -0,0 +1,104 @@
+        /// The view model for the sample.
+        @ObservedObject var model: Model
+        
+        /// The action to perform when the parameters are created, i.e, when the "Done" button is pressed.
+        let action: (ServiceVersionParameters) -> Void
+        
+        /// The action to dismiss the view.
+        @Environment(\.dismiss) private var dismiss
+        
+        /// The name of the version entered by the user.
+        @State private var versionName = ""
+        
+        /// The description of the version entered by the user.
+        @State private var versionDescription = ""
+        
+        /// The version access for the version selected by the user.
+        @State private var selectedVersionAccess: VersionAccess = .public
+        
+        var body: some View {
+            NavigationStack {
+                Form {
+                    TextField("Name", text: $versionName)
+                        .autocorrectionDisabled()
+                        .textInputAutocapitalization(.never)
+                        .onChange(of: versionName) { newVersionName in
+                            // Ensures the inputted version name is valid.
+                            let formattedVersionName = newVersionName
+                                .trimmingPrefix(" ")
+                                .replacing(/[.;'"]/, with: "")
+                                .prefix(62)
+                            
+                            self.versionName = String(formattedVersionName)
+                        }
+                    
+                    TextField("Description", text: $versionDescription )
+                    
+                    Picker("Access Permissions", selection: $selectedVersionAccess) {
+                        ForEach(VersionAccess.allCases, id: \.self) { versionAccess in
+                            Text(versionAccess.label)
+                        }
+                    }
+                }
+                .navigationTitle("New Version")
+                .navigationBarTitleDisplayMode(.inline)
+                .toolbar {
+                    ToolbarItem(placement: .cancellationAction) {
+                        Button("Cancel", role: .cancel) {
+                            model.clearSelection()
+                            dismiss()
+                        }
+                    }
+                    
+                    ToolbarItem(placement: .confirmationAction) {
+                        Button("Add") {
+                            let parameters = ServiceVersionParameters()
+                            parameters.access = selectedVersionAccess
+                            parameters.name = versionName
+                            parameters.description = versionDescription
+                            
+                            action(parameters)
+                            dismiss()
+                        }
+                        .disabled(versionName.isEmpty)
```

Comment:
If the name can include spaces, which it looks like it can, then the button should also be disabled when the last character is a space. That way, the user can still enter a space (followed by additional non-whitespace characters), but not be able to add while the name ends in a space.

Reply:
It looks like beginning and trailing spaces are actually trimmed when the version is made, and the errors I was getting testing those were due to the version already existing. (Not sure why the readme specified that it couldn't begin with a space, though.)

Reply:
In that case, you'll want to trim leading/trailing whitespace before verifying the character count. Theoretically, someone could be prevented from submitting a valid name because they have accidentally included leading/trailing whitespace that makes the name go over the character limit.

Reply:
@philium I am not sure how I would do this without trimming the whitespace, which would then prevent using spaces at all. Any suggestions?

Reply:
You would do it here:

```swift
.disabled(versionName.isEmpty || versionName.last!.isWhitespace)
```